### PR TITLE
feat: fetch image hashes when generating manifest

### PR DESCRIPTION
**Pull request by vtj2105**

### CODE REVIEW

File: test/generate-manifest.test.js [hunk 1]
```diff
@@ -82,4 +82,30 @@ describe('Generate Complete Manifest', function () {
...
+
+  it('should produce an error if a container contains an invalid image', async function () {
+    const manifest = JSON.parse(JSON.stringify(manifestJson))
+    manifest['manifest']['containers'][0]['image'] = 'hello-world@1231984'
+    await fse.writeJson(manifestMock, manifest)
+    const result = generateManifest(varsMock, manifestMock)
+    return expect(result).to.be.rejected
+  })
+
+  it('should resolve image tags to proper digest', async function () {
```
Comment (sharafian) [inline test/generate-manifest.test.js]: are these tests actually hitting the live docker hub? If so, we should try to mock out the network call instead

Comment (traviscrist) [reply test/generate-manifest.test.js]: @sharafian  normally I would agree but there could be some value in having this actually hit dockerhub acting like an integration test if it does change the api and stops working. 

Should we have some kind of docker integration test and a mocked version as well?

Comment (sharafian) [reply test/generate-manifest.test.js]: Yeah, I suppose it's ok in that case. Probably not worth splitting into integration tests for a repo this small but maybe in the future

File: src/generate-manifest.js [hunk 2]
```diff
@@ -58,17 +59,62 @@ const generateManifest = async function (codiusVarsPath, codiusPath) {
   }
 
   removeDescriptions(generatedManifest) // remove description fields from manifest
-  debug(`Generated Manifest: ${JSON.stringify(generatedManifest, null, 2)}`)
+
+  // check the digest of each container image
+  const containers = generatedManifest['manifest']['containers']
```
Comment (sharafian) [inline src/generate-manifest.js]: this variable isn't being used; can you run lint

File: test/generate-manifest.test.js (hunk 1 above)
Comment (traviscrist) [reply test/generate-manifest.test.js]: Lets keep it but add a comment to the test or put it in a section integration tests by itself so its clear this this relies on external resources to pass.

File: src/generate-manifest.js (hunk 2 above)
Comment (vtj2105) [reply src/generate-manifest.js]: The containers variable is being used in the for loop in the next line

File: test/generate-manifest.test.js (hunk 1 above)
Comment (traviscrist) [reply test/generate-manifest.test.js]: Looks good

**Pull request merged**